docs/components/index.md

```diff
@@ -0,0 +1,337 @@
+---
+title: Blazor components
+author: guardrex
+description: Learn how to create and use Blazor components, the fundamental building blocks of Blazor apps provided by compiled Razor or C# files.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/16/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/components/index
+---
+# Blazor components
+
+By [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnet/Blazor.Docs/docs/components/common/samples/) ([how to download](xref:client-side/blazor/index#view-and-download-samples)). See the [Get started](xref:client-side/blazor/get-started) topic for prerequisites.
+
+Blazor apps are built using *components*. A component is a self-contained chunk of user interface (UI), such as a page, dialog, or form. A component includes both the HTML markup to render along with the processing logic needed to inject data or respond to UI events. Typically, components are written using Razor files (*\*.cshtml*). Components are flexible and lightweight, and they can be nested, reused, and shared between projects.
+
+## Component classes
+
+Components are implemented using C# classes. Classes are usually saved in *\*.cshtml* files using a combination of HTML and C#.
+
+Dynamic rendering logic (for example, loops, conditionals, expressions) are added using an embedded C# syntax called [Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor).
+
+When a Blazor app is compiled, the HTML markup and C# rendering logic are converted into a component class. The name of the generated .NET class matches the name of the file.
+
+Members of the component class are defined in a `@functions` block (more than one `@functions` block is permissible). In the `@functions` block, component state (properties, fields) is specified along with methods for event handling or for defining other component logic.
+
+Component members can then be used as part of the component's rendering logic using C# expressions that start with `@`. For example, a C# field is rendered by prefixing `@` to the field name. The following example evaluates and renders:
+
+* `_headingFontStyle` to the CSS property value for `font-style`.
+* `_headingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@_headingFontStyle">@_headingText</h1>
+
+@functions {
+    private string _headingFontStyle = "italic";
+    private string _headingText = "Put on your new Blazor!";
+}
+```
+
+## Directives
+
+A component may specify one or more [directives](https://docs.microsoft.com/aspnet/core/mvc/views/razor#directives) at the top of the file.
```

Comment:
Razor directives don't actually have to be at the top of the file.

Replace with: Blazor supports the following Razor directives when authoring components:

Reply:
I've had trouble a couple of times in testing when, for example, a directive is preceded by a Razor comment (or HTML comment). It might have been a one-off thing tho. It may have also just been a RP scenario, where RP might be a bit fussier about it.